generationkwh/fareperiodcurve.py

```diff
@@ -7,17 +7,39 @@
 """
 
 import datetime
+from dateutil.relativedelta import relativedelta
 import numpy
 try:
     import libfacturacioatr
 except ImportError:
     libfacturacioatr = None
 
+DATE_START_NEW_FARES = datetime.date(2021,6,1)
 
 class FarePeriodCurve(object):
+
     def __init__(self, holidays):
         self.holidays = holidays
 
+    def get_pre_post_new_fares_periods(self, start, end):
+        if end < DATE_START_NEW_FARES or start >= DATE_START_NEW_FARES:
+            return [(start, end)]
+        translated_end = end - relativedelta(days=(end - DATE_START_NEW_FARES).days+1)
```

Comment:
I think you are "circumcoding" here (doing a greater walk for something that can be simpler). 

You